### PR TITLE
V2 meetings list

### PR DESCRIPTION
**Pull request by svenlombaert**
Added a proposal for the meetings.list endpoint

For now, all filters are required (because I need it like that).  Making filters optional is easy and backwards compatible. This endpoint is mainly needed for the dashboard.

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -951,6 +951,36 @@ Update a department
...
+
+Search the meetings
+
++ Request (application/json)
+
+    + Attributes (object)
+        + filter (object, required)
+            + employee_id: 1 (number, required) - Use this to filter by employee
+            + start_date: `2017-01-01` (string, required) - Format ISO 8601: yyyy-mm-dd
+            + end_date: `2017-01-14` (string, required) - Format ISO 8601: yyyy-mm-dd
```
Comment (Steerlin) [inline apiary.apib]: maybe add that the total period can not exceed X weeks ?

File: apiary.apib [hunk 2]
```diff
@@ -951,6 +951,36 @@ Update a department
...
+
++ Response 200 (application/json)
+
+    + Attributes (object)
+        + _items (array)
+            + (object)
+                + id: 102 (number)
+                + name: Sprint Planning (string)
```
Comment (Steerlin) [inline apiary.apib]: quotes

File: apiary.apib [hunk 3]
```diff
@@ -951,6 +951,36 @@ Update a department
...
+                + datetime: `2017-01-12T14:30:00+01:00` (string) - Datetime in ISO 8601 format: yyyy-mm-ddThh:mm:ss+|-hh:mm
+                + duration: `06:00` (string)
```
Comment (Steerlin) [inline apiary.apib]: duration should standard be in minutes ?

File: apiary.apib [hunk 4]
```diff
@@ -951,6 +951,36 @@ Update a department
...
+                + duration_timetracked: `02:25` (string)
+                + employee_id: 40 (number)
+                + company (object) - The meeting could take place at the address of a company
```
Comment (Steerlin) [inline apiary.apib]: there is no address in the company ?

File: apiary.apib (hunk 2 above)
Comment (svenlombaert) [inline apiary.apib]: doesn't need them

File: apiary.apib (hunk 3 above)
Comment (svenlombaert) [inline apiary.apib]: indeed, that's a mistake

File: apiary.apib [hunk 5]
```diff
@@ -1026,6 +1056,7 @@ Register a webhook.
                 + TIMETRACKING_DELETED
                 + INVOICE_ADDED
                 + INVOICE_UPDATED
+                + INVOICE_BOOKED
```
Comment (Steerlin) [inline apiary.apib]: _a random web-hook appears_

File: apiary.apib (hunk 4 above)
Comment (svenlombaert) [inline apiary.apib]: I don't need it for now, so I chose not to add it.

File: apiary.apib (hunk 5 above)
Comment (svenlombaert) [inline apiary.apib]: someone didn't run build before comitting! 🤓

File: src/meetings.apib [hunk 6]
```diff
@@ -1,5 +1,35 @@
...
+    + Attributes (object)
+        + _items (array)
+            + (object)
+                + id: 102 (number)
+                + name: Sprint Planning (string)
+                + datetime: `2017-01-12T14:30:00+01:00` (string) - Datetime in ISO 8601 format: yyyy-mm-ddThh:mm:ss+|-hh:mm
+                + duration: `06:00` (string)
+                + duration_timetracked: `02:25` (string)
+                + employee_id: 40 (number)
+                + company (object) - The meeting could take place at the address of a company
```
Comment (Steerlin) [inline src/meetings.apib]: if so at least the address should be included in the response ?

File: src/meetings.apib [hunk 7]
```diff
@@ -1,5 +1,35 @@
...
+
++ Response 200 (application/json)
+
+    + Attributes (object)
+        + _items (array)
+            + (object)
+                + id: 102 (number)
+                + name: Sprint Planning (string)
+                + datetime: `2017-01-12T14:30:00+01:00` (string) - Datetime in ISO 8601 format: yyyy-mm-ddThh:mm:ss+|-hh:mm
+                + duration: `06:00` (string)
```
Comment (Steerlin) [inline src/meetings.apib]: minutes ?

Comment (Steerlin) [inline src/meetings.apib]: this could be confused with 06 minutes and 00 seconds

**Comment by svenlombaert** [comment]
Actually, I embedded company (because I only need to know the company) but the meeting could be linked to a contact as well. So probably, I need to change this to `customer` and do something like this then:

```
customer: {
    id: 124
    type: 'contact'
    name: 'Freddy De Vadder'
}
```

File: apiary.apib (hunk 1 above)
Comment (svenlombaert) [inline apiary.apib]: I'll take then for now, and then we'll see

Comment (Steerlin) [inline apiary.apib]: guess you mean ten ?

Comment (svenlombaert) [inline apiary.apib]: That's what I wrote :trollface:

File: apiary.apib [hunk 8]
```diff
@@ -951,6 +951,41 @@ Update a department
...
+    + Attributes (object)
+        + _items (array)
+            + (object)
+                + id: 102 (number)
+                + name: Sprint Planning (string)
+                + scheduled_at: `2017-01-12T14:30:00+01:00` (string) - Datetime in ISO 8601 format: yyyy-mm-ddThh:mm:ss+|-hh:mm
+                + duration: 240 (number) - Duration in minutes
+                + duration_timetracked: 120 (number) - Duration in minutes
+                + done: true (boolean)
+                + employee_id: 40 (number)
```
Comment (jenssegers) [inline apiary.apib]: Embed employee here

Comment (svenlombaert) [inline apiary.apib]: done

File: apiary.apib [hunk 9]
```diff
@@ -951,6 +951,44 @@ Update a department
...
+        + _items (array)
+            + (object)
+                + id: 102 (number)
+                + name: Sprint Planning (string)
+                + description: `During this meeting we'll plan our new stories` (string)
+                + scheduled_at: `2017-01-12T14:30:00+01:00` (string) - Datetime in ISO 8601 format: yyyy-mm-ddThh:mm:ss+|-hh:mm
+                + duration: 240 (number) - Duration in minutes
+                + duration_timetracked: 120 (number) - Duration in minutes
+                + done: true (boolean)
+                + employee (object)
```
Comment (svenlombaert) [inline apiary.apib]: ditch this

**Pull request closed without merging**